### DISCUSSION TITLE
FAIR for Jupyter Notebooks: A Practical Guide

### DISCUSSION THREAD

**Discussion by Aleem2** [RSEAA/RSEAA.github.io#61] [Posters for #RSEAA2023]
**FAIR for Jupyter Notebooks: A practical guide**

[Learn why and how you should make your Jupyter Notebooks FAIR
](https://doi.org/10.6084/m9.figshare.24135960.v1)
Jupyter Notebooks have been widely used in research and data science [1]. But despite this, there are several challenges in making them findable, accessible, interoperable and reusable (FAIR) to the benefit of both the creators and users. A plausible approach is to apply the FAIR Principles to ensure the appropriate use of Jupyter Notebooks. However, the FAIR Principles are high-level aspirations and can be difficult to apply in practice. Also, translating the FAIR principles into practice will vary for each discipline.

This resource is also available as [ARDC web resource](https://ardc.edu.au/resource/fair-for-jupyter-notebooks-a-practical-guide/) and [github repository](https://github.com/Aleem2/FAIR4JupyterNotebook) as an example, and[ workflow](https://github.com/Aleem2/FAIR4JupyterNotebook/blob/main/Workflow-FAIR4JN.md) for quick adoption.

**Comment by supernord** [comment]
It's possible to consider a notebook as a computational workflow, in which case it could be registered and assigned a DOI 😄 
e.g. WorkflowHub https://workflowhub.eu/

**Reply by Aleem2** [reply]
Totally agree with you. Yes a DOI can be created and is recommended in publishing finalised work. I have created one for the FAIR for Jupyter Notebook. DOI: 10.5281/zenodo.7690164. DOI's also addresses Findable component well.

**Comment by SaranjeetKaur** [comment]
Making Jupyter Notebooks FAIR sounds cool!

**Comment by betsybookwyrm** [comment]
my fave tip for sharing jupyter notebooks and versioning them with git or other version control is to use https://github.com/kynan/nbstripout to clean up the .ipynb file before committing it. Using pre-commit https://github.com/kynan/nbstripout#using-nbstripout-as-a-pre-commit-hook you can set it up to automatically clean up the notebook every time you commit.

Any other tips and tricks from peoples' experience?

**Comment by betsybookwyrm** [comment]
Another fun thing to do with your jupyter notebooks is to make your slide presentations in them :D https://rise.readthedocs.io is an amazing tool, really great for workshops and presentations with code you want to run live. And then you can use Aleem's tips to share the whole thing :)

**Reply by Aleem2** [reply]
This is an excellent one. JNs are all about presentation of analysis. So this one will standout. Thanks for sharing this tip.

**Comment by SaranjeetKaur** [comment]
From the poster session Zoom chat:

Pradeep Rajasekhar to Everyone (14 Sep 2023, 11:07)
https://github.com/RSEAA/RSEAA.github.io/discussions/61
FAIR for Jupyter Notebooks: A practical guide
 
Pradeep Rajasekhar to Everyone (14 Sep 2023, 11:11)
https://figshare.com/articles/poster/FAIR_for_Jupyter_Notebooks_pdf/24135960/1
 
Betsy Alpert to Everyone (14 Sep 2023, 11:17)
https://choosealicense.com/

Andrew (Alf) Leahy to Everyone (14 Sep 2023, 11:19)
CITATION.cff cool. I learned something!  https://citation-file-format.github.io/
 
Betsy Alpert to Everyone (14 Sep 2023, 11:21)
yess the citation.cff thing is great! And then on github a little "cite this repository" thing appears on the side of your repository :)

Betsy Alpert to Everyone (14 Sep 2023, 11:21)
an example with a citation.cff - https://github.com/QUT-Digital-Observatory/coordination-network-toolkit
 
Betsy Alpert to Everyone (14 Sep 2023, 11:22)
note the "cite this repository" under the About section on the right, it's the best
 
Andrew (Alf) Leahy to Everyone (14 Sep 2023, 11:22)
https://ardc.edu.au/resource/fair-for-jupyter-notebooks-a-practical-guide/ … should we be capturing these in persistent discussion somewhere?
 
Aleem Uddin to Everyone (14 Sep 2023, 11:23)
https://ardc.edu.au/resource/fair-for-jupyter-notebooks-a-practical-guide/
 
Paul Leopardi to Everyone (14 Sep 2023, 11:23)
I have recently created a related poster at https://forum.access-hive.org.au/t/poster-improving-the-reusability-of-access-model-input-data/1150
 
Betsy Alpert to Everyone (14 Sep 2023, 11:24)
Thanks Aleem, I will definitely be sharing your resources with people
awesome! I really need to add RO Crate scaffolding to my bits and pieces. For those who don't know RO Crate - https://www.researchobject.org/ro-crate/
 
Paul Leopardi to Everyone (14 Sep 2023, 11:29)
Galaxy: https://scholarworks.iu.edu/dspace/bitstream/handle/2022/18502/Provenance_071614.pdf?sequence=1&isAllowed=y

**Reply by Aleem2** [reply]
Any comments on improving the FAIR for Jupyter Notebook are welcome. I am happy to collaborate as well. Thanks for listening to the poster.
